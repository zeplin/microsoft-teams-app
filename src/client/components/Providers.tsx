import React, { FunctionComponent } from "react";
import {
    Provider,
    teamsDarkTheme,
    teamsHighContrastTheme,
    teamsTheme,
    ThemePrepared
} from "@fluentui/react-northstar";

const getTheme = (key?: string): ThemePrepared => {
    switch (key) {
        case "dark":
            return teamsDarkTheme;
        case "contrast":
            return teamsHighContrastTheme;
        default:
            return teamsTheme;
    }
};

interface ProvidersProps {
    theme?: string;
}

export const Providers: FunctionComponent<ProvidersProps> = ({ children, theme }) => (
    <Provider theme={getTheme(theme)}>
        {children}
    </Provider>
);
