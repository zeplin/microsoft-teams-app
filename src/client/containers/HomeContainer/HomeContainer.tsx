import React, { FunctionComponent, useEffect } from "react";
import { Loader } from "@fluentui/react-northstar";
import { useRouter } from "next/router";

import { storage, url } from "../../lib";
import { app } from "@microsoft/teams-js";

export const HomeContainer: FunctionComponent = () => {
    const {
        query: {
            channel,
            id,
            resourceName,
            resourceType,
            theme
        },
        replace
    } = useRouter();

    const getUrl = (): string => {
        if (!storage.getAccessToken()) {
            return url.getLoginUrl(id
                ? {
                    channel: channel as string,
                    id: id as string,
                    resourceName: resourceName as string,
                    resourceType: resourceType as string,
                    theme: theme as string
                }
                : {
                    channel: channel as string,
                    theme: theme as string
                });
        }
        if (id) {
            return url.getConfigurationUpdateUrl({
                channel: channel as string,
                id: id as string,
                resourceName: resourceName as string,
                resourceType: resourceType as string,
                theme: theme as string
            });
        }
        return url.getConfigurationCreateUrl({
            channel: channel as string,
            theme: theme as string
        });
    };

    useEffect(() => {
        console.log("useEffect", channel);

        app.initialize().then(() => {
            app.getContext().then(context => {
                console.log(JSON.stringify(context));
            });
        });
        // Workaround: Microsoft Teams initialize two iframes
        // One with real one and one with template url
        // TODO: Find a robust solution to disable second iframe
        if (channel && channel !== "{channelName}") {
            replace(getUrl());
        }
    }, []);

    return <Loader styles={{ height: "100vh" }} />;
};
