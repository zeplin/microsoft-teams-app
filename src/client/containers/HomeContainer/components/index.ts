export * from "./Login";
export * from "./Configuration";
