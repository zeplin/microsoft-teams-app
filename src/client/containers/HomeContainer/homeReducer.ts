import { Reducer } from "react";

import { Resource } from "../../requester";

type State = {
    status: Status.LOADING | Status.LOGIN;
} | {
    status: Status.CONFIGURATION;
    accessToken: string;
    selectedWorkspace: undefined;
    selectedResource: undefined;
} | {
    status: Status.CONFIGURATION;
    accessToken: string;
    selectedWorkspace: string;
    selectedResource: Resource | undefined;
}

type Action = {
    type: ActionType.COMPLETE_LOADING;
} | {
    type: ActionType.GET_TOKEN | ActionType.SET_SELECTED_WORKSPACE;
    value: string;
} | {
    type: ActionType.SET_SELECTED_RESOURCE;
    value: Resource | undefined;
}

export enum Status {
    LOADING,
    LOGIN,
    CONFIGURATION
}

export enum ActionType {
    COMPLETE_LOADING,
    GET_TOKEN,
    SET_SELECTED_WORKSPACE,
    SET_SELECTED_RESOURCE
}

export const homeReducer: Reducer<State, Action> = (state, action) => {
    switch (action.type) {
        case ActionType.COMPLETE_LOADING:
            return { status: Status.LOGIN };
        case ActionType.GET_TOKEN:
            return {
                status: Status.CONFIGURATION,
                accessToken: action.value,
                selectedWorkspace: undefined,
                selectedResource: undefined
            };
        case ActionType.SET_SELECTED_WORKSPACE:
            return {
                ...state,
                selectedWorkspace: action.value,
                selectedResource: undefined
            };
        case ActionType.SET_SELECTED_RESOURCE:
            return {
                ...state,
                selectedResource: action.value
            };
        default:
            throw new Error();
    }
};
