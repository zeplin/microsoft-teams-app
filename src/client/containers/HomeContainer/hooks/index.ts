export * from "./useHomeReducer";
export * from "./useConfigurationSave";
export * from "./useConfigurationDelete";
export * from "./useInitialize";
export * from "./useResources";
export * from "./useValidate";
export * from "./useStateUpdateFromConfiguration";
export * from "./useWorkspaces";
export * from "./useAuthenticate";
