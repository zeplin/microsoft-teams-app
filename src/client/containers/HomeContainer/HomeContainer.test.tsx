import React from "react";
import { HomeContainer } from "./HomeContainer";
import * as microsoftTeams from "@microsoft/teams-js";
import { fireEvent, render, RenderResult } from "@testing-library/react";
import { Providers } from "../../Providers";

jest.mock("@microsoft/teams-js", () => ({
    initialize: jest.fn(callback => callback()),
    appInitialization: {
        notifySuccess: jest.fn()
    },
    settings: {
        registerOnSaveHandler: jest.fn(),
        registerOnRemoveHandler: jest.fn(),
        setValidityState: jest.fn(),
        setSettings: jest.fn(),
        getSettings: jest.fn(callback => callback({ entityId: "" }))
    },
    authentication: {
        authenticate: jest.fn(({ successCallback }) => successCallback("accessToken"))
    }
}));

jest.mock("next/router", () => ({
    useRouter: (): object => ({
        query: {
            channel: "channel"
        }
    })
}));

function renderHome(): RenderResult {
    return render(
        <Providers>
            <HomeContainer />
        </Providers>
    );
}

describe("HomeContainer", () => {
    it("should render initially", () => {
        const spy = jest.spyOn(microsoftTeams, "initialize")
            .mockImplementation(() => undefined);

        const { container: { firstChild: component } } = renderHome();

        expect(component).toMatchSnapshot();

        spy.mockImplementation(callback => callback());
    });

    it("should initialize Microsoft Teams", () => {
        const spy = jest.spyOn(microsoftTeams.appInitialization, "notifySuccess");

        const { container: { firstChild: component } } = renderHome();

        expect(component).toMatchSnapshot();
        expect(spy).toBeCalledWith();
    });

    it("should change ui when login flow completed", () => {
        const spy = jest.spyOn(microsoftTeams.authentication, "authenticate");

        const { container: { firstChild: component }, getByText } = renderHome();

        expect(spy).not.toBeCalled();

        fireEvent.click(getByText("Log in Zeplin"));

        expect(component).toMatchSnapshot();
        expect(spy).toBeCalled();
    });
});
