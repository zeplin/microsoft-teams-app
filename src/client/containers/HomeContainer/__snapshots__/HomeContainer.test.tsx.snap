// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer should change ui when login flow completed 1`] = `
<div
  class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak"
  data-uses-unhanded-props="true"
  dir="ltr"
>
  <div
    class="ui-flex am ao bq br bs"
    data-uses-unhanded-props="true"
  >
    <div />
    <div
      class="ui-flex am ao bq br fe"
      data-uses-unhanded-props="true"
    >
      <span
        class="ui-text bu"
        data-uses-unhanded-props="true"
      >
        You are connecting 
        <span
          class="ui-text ff"
          data-uses-unhanded-props="true"
        >
          #
          channel
        </span>
         to Zeplin.
      </span>
      <span
        class="ui-text"
        data-uses-unhanded-props="true"
      >
        Can
        '
        t find the project/styleguide to connect to?
         
        <a
          class="ui-text fg fh fi"
          data-uses-unhanded-props="true"
          dir="auto"
          href="https://zpl.io/msteams-app-docs"
          target="_blank"
        >
          Learn more about permissions.
        </a>
      </span>
      <div
        class="ui-flex am bq br fj"
        data-uses-unhanded-props="true"
      >
        <div
          class="ui-flex__item fk fl fm"
        >
          <div
            class="ui-dropdown"
          >
            <div
              class="ui-dropdown__container am fn bb cq cr cs ct cu cv cw cx cl bq fo fp fq fr fs ft fu fv fw fx fy fz ga gb"
            >
              <div
                class="ui-dropdown__selected-items am fn gc gd bq ge"
              >
                <button
                  aria-haspopup="true"
                  aria-labelledby="dropdown-trigger-button-1"
                  class="ui-button bx by gf b gg cc cd gh al bb ce gi cg gj ci cj ck cl cm cn co cp cq cr cs ct cu cv cw cx gk gl gm fw fx fy fz db gn go gp gq gr gs dd gt gu de gv gw gx gy gz ha hb hc hd he hf hg hh hi hj hk hl hm hn ho hp hq hr hs ht hu df dg dh di dj dk dl dm dn do dp dq dr ds dt du hv hw hx hy dz hz eb ec ed ee ef eg eh ei ej ek el em en eo ep ia ib ic id eu ie ew ex ey if ig ih ii ij ik il im in io bq ar ip iq ir is ui-dropdown__trigger-button"
                  data-aa-class="Button"
                  data-toggle="true"
                  data-uses-unhanded-props="true"
                  id="dropdown-trigger-button-1"
                  role="button"
                  type="button"
                >
                  <span
                    class="ui-button__content ar ez fb fc bu fd"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Select Workspace
                  </span>
                </button>
              </div>
              <div
                aria-hidden="true"
                class="ui-box al am an it iu iv cj iw ip iq ir is ix iy br iz ui-dropdown__toggle-indicator"
                data-aa-class="Box"
                data-uses-unhanded-props="true"
                role="img"
              />
              <ul
                aria-hidden="true"
                aria-labelledby="downshift-0-label"
                class="ui-list ba ja ce jb cg jc ip iq ir is cl cq cr cs ct fo fp fq fr cu cv cw cx jd je gc bq a jf ui-dropdown__items-list"
                data-aa-class="List"
                data-uses-unhanded-props="true"
                id="downshift-0-menu"
                role="listbox"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
        <div
          class="ui-flex__item fk fl fm"
        >
          <div
            class="ui-dropdown"
          >
            <div
              class="ui-dropdown__container am fn bb cq cr cs ct jg jh ji jj cl bq fo fp fq fr fs jk fu fv jl gb iw jm"
            >
              <div
                class="ui-dropdown__selected-items am fn gc gd bq ge"
              >
                <button
                  aria-disabled="true"
                  aria-haspopup="true"
                  aria-labelledby="dropdown-trigger-button-2"
                  class="ui-button bx by gf jn gg cc cd gh al bb ce gi cg gj ci jo ck cl cm cn co cp cq cr cs ct cu cv cw cx gk gl gm fw fx fy fz db gn go gp gq gr gs dd gt gu de gv gw gx gy gz ha hb hc hd he hf hg hh hi hj hk hl hm hn ho hp hq hr hs ht hu df dg dh di dj dk dl dm dn do dp dq dr ds dt du hv hw hx hy dz hz eb ec ed ee ef eg eh ei ej ek el em en eo ep ia ib ic id eu ie ew ex ey if ig ih ii ij ik il im in io jp bq ar ip iq ir is ui-dropdown__trigger-button"
                  data-aa-class="Button"
                  data-toggle="true"
                  data-uses-unhanded-props="true"
                  disabled=""
                  id="dropdown-trigger-button-2"
                  role="button"
                  type="button"
                >
                  <span
                    class="ui-button__content ar ez fb fc bu fd"
                    data-uses-unhanded-props="true"
                    dir="auto"
                  >
                    Select Project/Styleguide
                  </span>
                </button>
              </div>
              <div
                aria-hidden="true"
                class="ui-box al am an jq iu iv jo iw ip iq ir is ix iy br iz ui-dropdown__toggle-indicator"
                data-aa-class="Box"
                data-uses-unhanded-props="true"
                role="img"
              />
              <ul
                aria-hidden="true"
                aria-labelledby="downshift-1-label"
                class="ui-list ba ja ce jb cg jc ip iq ir is cl cq cr cs ct fo fp fq fr cu cv cw cx jd je gc bq a jf ui-dropdown__items-list"
                data-aa-class="List"
                data-uses-unhanded-props="true"
                id="downshift-1-menu"
                role="listbox"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui-flex am ao bq br bt"
      data-uses-unhanded-props="true"
    >
      <span
        class="ui-text bu"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        Select the events you want to get a message for:
      </span>
      <div>
        <div
          class="ui-flex am bq br fj"
          data-uses-unhanded-props="true"
        >
          <div
            class="ui-flex am ao bq br bt"
            data-uses-unhanded-props="true"
          >
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Screens
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Screens added
                    </span>
                  </div>
                </span>
              </div>
            </div>
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Screen versions
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Versions added to screen
                    </span>
                  </div>
                </span>
              </div>
            </div>
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Components
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Components added, removed or updated on Styleguide
                    </span>
                  </div>
                </span>
              </div>
            </div>
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Colors
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Colors added, removed or updated on Styleguide
                    </span>
                  </div>
                </span>
              </div>
            </div>
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Text styles
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Text styles added, removed or updated on Styleguide
                    </span>
                  </div>
                </span>
              </div>
            </div>
          </div>
          <div
            class="ui-flex am ao bq br bt"
            data-uses-unhanded-props="true"
          >
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Spacing tokens
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Spacing tokens added, removed or updated on Styleguide
                    </span>
                  </div>
                </span>
              </div>
            </div>
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Notes
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Note added
                    </span>
                  </div>
                </span>
              </div>
            </div>
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Replies
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Replied to note
                    </span>
                  </div>
                </span>
              </div>
            </div>
            <div>
              <div
                aria-checked="true"
                aria-disabled="false"
                class="ui-checkbox bb jr js jt cj cl b ju jv jw jx ci jy de df dg dh di dj dk dl dm dn do dp jz ka kb kc du dv dw dx dy dz ea eb ec kd ee ef eg eh ei ej ek ke kf kg kh ep eq er es et eu ev ew ex ki kj kk kl km kn ko"
                data-aa-class="Checkbox"
                data-is-focusable="true"
                data-uses-unhanded-props="true"
                role="checkbox"
                tabindex="0"
              >
                <div
                  class="ui-box kp kq kr ks kt ku kv kw kx ky cq cr cs ct fu kz la lb lc ld le iq ir is lf lg lh li iw lj iu iv ui-checkbox__indicator"
                  data-uses-unhanded-props="true"
                />
                <span
                  class="ui-text ba lk ll ui-checkbox__label"
                  data-uses-unhanded-props="true"
                >
                  <div
                    class="ui-flex am ao bq br"
                    data-uses-unhanded-props="true"
                  >
                    <span
                      class="ui-text"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Members
                    </span>
                    <span
                      class="ui-text lm bw"
                      data-uses-unhanded-props="true"
                      dir="auto"
                    >
                      Members invited
                    </span>
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HomeContainer should initialize Microsoft Teams 1`] = `
<div
  class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak"
  data-uses-unhanded-props="true"
  dir="ltr"
>
  <div
    class="ui-flex am ao bq br bs"
    data-uses-unhanded-props="true"
  >
    <div />
    <div
      class="ui-flex am ao bq br bt"
      data-uses-unhanded-props="true"
    >
      <span
        class="ui-text bu bv bw"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        One more step before finishing connection
      </span>
      <span
        class="ui-text"
        data-uses-unhanded-props="true"
        dir="auto"
      >
        Once proceeding, we will ask you to authorize Microsoft Teams access through a separate window. Once it’s authorized, you can do more advanced settings on the Teams channel linked with Zeplin.
      </span>
    </div>
    <div>
      <button
        class="ui-button bx by bz ca cb cc cd an al bb ce cf cg ch ci cj ck cl cm cn co cp cq cr cs ct cu cv cw cx cy cz da db dc dd de df dg dh di dj dk dl dm dn do dp dq dr ds dt du dv dw dx dy dz ea eb ec ed ee ef eg eh ei ej ek el em en eo ep eq er es et eu ev ew ex ey"
        data-aa-class="Button"
        data-uses-unhanded-props="true"
      >
        <span
          class="ui-button__content ar ez fb fc bu fd"
          data-uses-unhanded-props="true"
          dir="auto"
        >
          Log in Zeplin
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`HomeContainer should render initially 1`] = `
<div
  class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak"
  data-uses-unhanded-props="true"
  dir="ltr"
>
  <div
    class="ui-loader al am an ao"
    data-aa-class="Loader"
    data-uses-unhanded-props="true"
    id="loader-home"
    role="progressbar"
  >
    <div
      class="ui-box ap aq ar ui-loader__indicator"
      data-uses-unhanded-props="true"
    >
      <div
        class="ui-box as at au av aw ax ay az ba ar bb bc bd be bf bg bh bi bj bk bl bm bn bo bp ui-loader__svg"
        data-uses-unhanded-props="true"
      />
    </div>
  </div>
</div>
`;
